[Docs] add set command examplee to sql api (#18027)
apache/druid: docs/multi-stage-query/reference.md

Keep the following in mind when using EXTERN to export rows:
- Only INSERT statements are supported.
- Only `CSV` format is supported as an export format.
- Partitioning (PARTITIONED BY) and clustering (CLUSTERED BY) aren't supported with EXTERN statements.
- You can export to Amazon S3, Google GCS, or local storage.
- The destination provided should contain no other files or directories.

> When you export data, use SET to restrict `rowsPerPage` to control the size of exported files. For example: 

Reviewer: I think the original text is better suited to generally introduce the parameter and why you want to use it (not jumping into how you want to set it)

> 

Reviewer: ```suggestion
The following statement shows the format of a SQL query using EXTERN to export rows:

```

```sql
SET rowsPerPage=<number_of_rows>;

> 

Reviewer: ```suggestion
```

FROM <table>
```

> When the number of rows in the result set exceeds the value of the parameter, Druid splits the output into multiple files.

Reviewer: Move this line back above (see previous comment)

> For details about SET, see [SET statements](../querying/sql.md#set-statements).

Reviewer: ```suggestion
For details about applying context parameters using SET, see [SET statements](../querying/sql.md#set-statements).
```

- All join types are supported with `sortMerge`: LEFT, RIGHT, INNER, FULL, and CROSS.

> The following example shows a single sort-merge join stage where it explicitly set `sqlJoinAlgorithm` to `sortMerge` using the SET command. This query also takes `eventstream` (partitioned on `user_id`) and `users` (partitioned on id) as `inputs`, with no limit on the size of either input.

Reviewer: ```suggestion
The following query runs a single sort-merge join stage that takes the following inputs:
* `eventstream` partitioned on `user_id`
* `users` partitioned on `id`

There is no limit on the size of either input.
The SET clause sets the `sqlJoinAlgorithm` context parameter so that Druid applies the sort-merge join algorithm for the query.
```

```sql
SET sqlJoinAlgorithm='sortMerge';

> 

Reviewer: ```suggestion
```
